### PR TITLE
regularize EPL at origin

### PR DESCRIPTION
**Pull request by dangilman**
I just implemented the property I want in the deflection angles alpha(0, 0) = 0 by replacing nan values with zero. The function can only be nan at the origin so it works

### CODE REVIEW

File: lenstronomy/LensModel/Profiles/epl.py [hunk 1]
```diff
@@ -183,8 +183,27 @@ class EPLMajorAxis(LensProfileBase):
...
     def __init__(self):
+
         super(EPLMajorAxis, self).__init__()
 
+    @staticmethod
+    def _regularize(alpha):
+
+        alpha_real, alpha_imag = alpha.real, alpha.imag
+
+        if isinstance(alpha_real, float) or isinstance(alpha_real, int):
```
Comment (sibirrer) [inline lenstronomy/LensModel/Profiles/epl.py:194]: can you simplify this with this functionality in numpy? https://numpy.org/doc/stable/reference/generated/numpy.nan_to_num.html

File: lenstronomy/LensModel/Profiles/epl.py [hunk 2]
```diff
@@ -214,7 +233,9 @@ def derivatives(self, x, y, b, t, q):
         alpha = 2/(1+q)*(b/R)**t*R_omega
 
         # return real and imaginary part
-        return alpha.real, alpha.imag
+        alpha_real, alpha_imag = self._regularize(alpha)
```
Comment (sibirrer) [inline lenstronomy/LensModel/Profiles/epl.py:236]: isn't this also an issue for derivatives and Hessians?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Could you simplify things with `numpy.nan_to_num`? I don't see a specific test case that checks the case of turning nan's into zero. Thanks!

**Comment by dangilman** [comment]
I added the numpy function. Passing in (x, y) = (0, 0) creates np.nan values inside the class that get output as zeros, but I added another test function that tests the new regularize function explicitly. And it is also an issue for hessian, but hessian is computed by calling derivatives so it is taken care of

**Comment by dangilman** [comment]
nevermind, was looking at the wrong hessian class.... let me fix it for the hessian also

**Comment by dangilman** [comment]
not really sure what to do with infinities... I just default them to plus/minus 1000

File: lenstronomy/LensModel/Profiles/epl.py [hunk 3]
```diff
@@ -183,8 +183,18 @@ class EPLMajorAxis(LensProfileBase):
...
 
     def __init__(self):
+
         super(EPLMajorAxis, self).__init__()
 
+    @staticmethod
+    def _regularize(value):
+
+        out = np.nan_to_num(value, posinf=1e+3, neginf=-1e+3)
+        if isinstance(value, float) or isinstance(value, int):
```
Comment (sibirrer) [inline lenstronomy/LensModel/Profiles/epl.py:193]: do you need this part?

File: lenstronomy/LensModel/Profiles/epl.py [hunk 4]
```diff
@@ -183,8 +183,18 @@ class EPLMajorAxis(LensProfileBase):
     param_names = ['b', 't', 'q', 'center_x', 'center_y']
 
     def __init__(self):
+
         super(EPLMajorAxis, self).__init__()
 
+    @staticmethod
+    def _regularize(value):
+
+        out = np.nan_to_num(value, posinf=1e+3, neginf=-1e+3)
```
Comment (sibirrer) [inline lenstronomy/LensModel/Profiles/epl.py:192]: posinf and neginf do not need to be specified I think.

File: test/test_LensModel/test_Profiles/test_epl.py [hunk 5]
```diff
@@ -100,6 +100,42 @@ def test_static(self):
...
+        npt.assert_almost_equal(f_yy, 0.)
+
+        x = 0.3
+        y = 0.1
+        R = np.sqrt(x ** 2 + y ** 2)
+        f_x, f_y = self.EPL.derivatives(x, y, phi_E, e1, e2, gamma)
+        npt.assert_almost_equal(f_x, phi_E * x/R)
+        npt.assert_almost_equal(f_y, phi_E * y/R)
+
+        value = np.nan
```
Comment (sibirrer) [inline test/test_LensModel/test_Profiles/test_epl.py:130]: I was more thinking of putting in (0,0) and see whether you get regular behavior out

**Review by sibirrer — CHANGES_REQUESTED** [review]
I don't think you need the array and float/int distinction with the new routine. This will simplify things quite a bit.

File: test/test_LensModel/test_Profiles/test_epl.py (hunk 5 above)
Comment (dangilman) [reply test/test_LensModel/test_Profiles/test_epl.py:130]: I do that on line 111

File: lenstronomy/LensModel/Profiles/epl.py (hunk 4 above)
Comment (dangilman) [reply lenstronomy/LensModel/Profiles/epl.py:192]: I'll take them out

File: lenstronomy/LensModel/Profiles/epl.py (hunk 3 above)
Comment (dangilman) [reply lenstronomy/LensModel/Profiles/epl.py:193]: It serves only to keep the input and output data types the same. If you supply a float, without that check you get back a numpy array. I can take it out if you want

Comment (sibirrer) [reply lenstronomy/LensModel/Profiles/epl.py:193]: I had guessed that the nan_to_num keeps the same in and output format. It's worth testing and if confirmed, you don't need this.

**Review by sibirrer — APPROVED** [review]

**Comment by sibirrer** [comment]
Thank you!

**Pull request merged**